src/ds/d400/d400-private.cpp

```diff
@@ -290,6 +290,76 @@ namespace librealsense
             return calc_intrinsic;
         }
 
+        //D405 needs special calculation because the ISP crops the full sensor image using non linear transformation.
+        rs2_intrinsics get_d405_color_stream_intrinsic(const std::vector<uint8_t>& raw_data, uint32_t width, uint32_t height)
+        {
+            // Convert normalized focal lenght and principal point to pixel units (K matrix format)
+            auto k_to_pixels = []( float3x3 & k, ds_rect_resolutions res )
+            {
+                k( 0, 0 ) = k( 0, 0 ) * resolutions_list[res].x / 2.f;  // fx
+                k( 1, 1 ) = k( 1, 1 ) * resolutions_list[res].y / 2.f;  // fy
+                k( 2, 0 ) = ( k( 2, 0 ) + 1 ) * resolutions_list[res].x / 2.f;  // ppx
+                k( 2, 1 ) = ( k( 2, 1 ) + 1 ) * resolutions_list[res].y / 2.f;  // ppy
+            };
+
+            // Scale focal lenght and principal point in pixel units from one resolution to another (K matrix format)
+            auto scale_pixel_k = []( float3x3 & k, ds_rect_resolutions in_res, ds_rect_resolutions out_res )
+            {
+                float scale_x = resolutions_list[out_res].x / static_cast< float >( resolutions_list[in_res].x );
+                float scale_y = resolutions_list[out_res].y / static_cast< float >( resolutions_list[in_res].y );
+                float scale = max( scale_x, scale_y );
+                float shift_x = ( resolutions_list[in_res].x * scale - resolutions_list[out_res].x ) / 2.f;
+                float shift_y = ( resolutions_list[in_res].y * scale - resolutions_list[out_res].y ) / 2.f;
+
+                k( 0, 0 ) = k( 0, 0 ) * scale;  // fx
+                k( 1, 1 ) = k( 1, 1 ) * scale;  // fy
+                k( 2, 0 ) = k( 2, 0 ) * scale - shift_x;  // ppx
+                k( 2, 1 ) = k( 2, 1 ) * scale - shift_y;  // ppy
+            };
+
+            auto table = check_calib< ds::d400_rgb_calibration_table >( raw_data );
+            auto raw_res = width_height_to_ds_rect_resolutions( 1280, 800 );
```

Comment:
Can you explain why we take this resolution?

Reply:
This is the raw sensor resolution.
These values where also hard coded in the matlab algorithm I was based on. Special for D405 handling.